singlestoredb/management/files.py

```diff
@@ -463,3 +470,577 @@ def __str__(self) -> str:
+
+            self.remove(new_path)
+
+        self._manager._patch(
+            f'files/fs/{self._location}/{old_path}',
+            json=dict(newPath=new_path),
+        )
+
+        return self.info(new_path)
+
+    def info(self, path: PathLike) -> FilesObject:
+        """
+        Return information about a file location.
+
+        Parameters
+        ----------
+        path : Path or str
+            Path to the file
+
+        Returns
+        -------
+        FilesObject
+
+        """
+        res = self._manager._get(
+            re.sub(r'/+$', r'/', f'files/fs/{self._location}/{path}'),
+            params=dict(metadata=1),
+        ).json()
+
+        return FilesObject.from_dict(res, self)
+
+    def exists(self, path: PathLike) -> bool:
+        """
+        Does the given file path exist?
+
+        Parameters
+        ----------
+        path : Path or str
+            Path to file object
+
+        Returns
+        -------
+        bool
+
+        """
+        try:
+            self.info(path)
+            return True
+        except ManagementError as exc:
+            if exc.errno == 404:
+                return False
+            raise
+
+    # TODO: remove from FileLocation?
+    def is_dir(self, path: PathLike) -> bool:
+        """
+        Is the given file path a directory?
+
+        Parameters
+        ----------
+        path : Path or str
+            Path to file object
+
+        Returns
+        -------
+        bool
+
+        """
+        try:
+            return self.info(path).type == 'directory'
+        except ManagementError as exc:
+            if exc.errno == 404:
+                return False
+            raise
+
+    # TODO: remove from FileLocation?
+    def is_file(self, path: PathLike) -> bool:
+        """
+        Is the given file path a file?
+
+        Parameters
+        ----------
+        path : Path or str
+            Path to file object
+
+        Returns
+        -------
+        bool
+
+        """
+        try:
+            return self.info(path).type != 'directory'
+        except ManagementError as exc:
+            if exc.errno == 404:
+                return False
+            raise
+
+    def _list_root_dir(self) -> List[str]:
+        """
+        Return the names of files in the root directory.
+        Parameters
+        ----------
+        """
+        res = self._manager._get(
+            f'files/fs/{self._location}',
+        ).json()
+        return [x['path'] for x in res['content'] or []]
+
+    # TODO: remove from FileLocation?
+    def listdir(
+        self,
+        path: PathLike = '/',
+        *,
+        recursive: bool = False,
+    ) -> List[str]:
+        """
+        List the files / folders at the given path.
+
+        Parameters
+        ----------
+        path : Path or str, optional
+            Path to the file location
+
+        Returns
+        -------
+        List[str]
+
+        """
+        if path == '' or path == '/':
```

Comment:
You may want to put the ManagementError before this block, or add a ManagementError to the `_list_root_dir` method so that they give consistent errors.

Reply:
Ok, your code is fine then.